x-pack/plugins/observability_solution/observability_ai_assistant/server/functions/execute_connector.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FunctionRegistrationParameters } from '.';
+import { FunctionVisibility } from '../../common';
+
+export function registerExecuteConnectorFunction({
+  functions,
+  resources,
+}: FunctionRegistrationParameters) {
+  functions.registerFunction(
+    {
+      name: 'execute_connector',
+      contexts: ['core'],
+      description: 'Use this function when user explicitly asks to call a kibana connector.',
+      visibility: FunctionVisibility.AssistantOnly,
+      parameters: {
+        type: 'object',
+        properties: {
+          id: {
+            type: 'string',
+            description: 'The id of the connector',
+          },
+          params: {
+            type: 'object',
+            description: 'The connector parameters',
+          },
+        },
+        required: ['id', 'params'],
+      } as const,
+    },
+    async ({ arguments: { id, params } }, signal) => {
+      const actionsClient = await (
+        await resources.plugins.actions.start()
+      ).getActionsClientWithRequest(resources.request);
+      await actionsClient.execute({ actionId: id, params });
+      return { content: {} };
```

Comment:
should `content` include whatever is returned from `actionsClient.execute` (the response body hopefully)?

Reply:
added in https://github.com/elastic/kibana/pull/179980/commits/f72d35a8ff288268f671339b4c07cdd780fafd71